src/components/ZoomableImage.vue

```diff
@@ -0,0 +1,90 @@
+<script setup>
+import { defineProps, ref} from 'vue'
+// eslint-disable-next-line no-unused-vars
+const props = defineProps(['imageSrc'])
+const image = ref(null)
+const imageLeftPos = ref(0)
+const imageTopPos = ref(0)
+const currentZoom = ref(1)
+
+const MIN_ZOOM = 1
+const MAX_ZOOM = 3
+const STEP_SIZE = 0.1
+let mouseX
+let mouseY
+
+const zoom = (e, direction) => {
+	if (!direction) direction = e.deltaY > 0 ? -1 : 1
+	let newZoom = currentZoom.value + direction * STEP_SIZE 
+	if (newZoom > MIN_ZOOM && newZoom < MAX_ZOOM) currentZoom.value = newZoom
+}
+
+// save the position clicked and add the pointermove listener
```

Comment:
when the user clicks on the image we want to start listening for a mouse movement event